### PR TITLE
feat(perception_online_evaluator): add perception_online_evaluator

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

Add perception_evaluator

This module allows for the evaluation of how accurately perception results are generated without the need for annotations. It is capable of confirming performance and can evaluate results from a few seconds prior, enabling online execution.

current metrics are

- Calculates lateral deviation between the predicted path and the actual traveled trajectory.

blue arrow are old predicted path and red arrows are actual history path the objects followed.
image

- Calculates lateral deviation between the smoothed traveled trajectory and the perceived position to evaluate the stability of lateral position recognition.
- Calculates yaw deviation between the smoothed traveled trajectory and the perceived position to evaluate the stability of yaw recognition.

smoothing path (window size=11)
image

rqt reconfigure
image

## Tests performed

psim (max of time_horizonx=5)

https://github.com/autowarefoundation/autoware.universe/assets/39142679/23ba4edc-d5d4-477b-835f-9ad52cb587a0

psim with rosbag(perception replyer) (max of time_horizonx=10)

https://github.com/autowarefoundation/autoware.universe/assets/39142679/b26e8603-aea8-4070-b4ab-34747d967ad0

logging simulator

image

evaluator_description: feat/perception_evaluator
2024/02/27 https://evaluation.tier4.jp/evaluation/reports/1542d256-59b7-54c2-8d54-fc960508f9eb/?project_id=prd_jt

evaluator_description: feat/perception_evaluator  time horizon 5
2024/02/27 https://evaluation.tier4.jp/evaluation/reports/ba008b2b-0242-52db-b18e-62781f6b554e/?project_id=prd_jt

evaluator_description: feat/perception_evaluator
2024/02/28 https://evaluation.tier4.jp/evaluation/reports/bbaa9fdc-cd30-56ef-b903-8e2908b1f674/?project_id=prd_jt

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: evaluator/perception_evaluator/src/utils/marker_utils.cpp
Comment (YoshiRi) [inline evaluator/perception_evaluator/src/utils/marker_utils.cpp:115]: Please fix build error before PR. This `id` is not used.

Comment (kosuke55) [reply evaluator/perception_evaluator/src/utils/marker_utils.cpp]: oh, thanks. fixed build werror in https://github.com/autowarefoundation/autoware.universe/pull/6493/commits/1d3e81021128f96290def911cb91f1a544d9daf1

**Comment by kosuke55** [comment]
rename to `perception_online_evaluator` because it is confusing if considering https://github.com/tier4/driving_log_replayer/blob/develop/driving_log_replayer/scripts/perception_evaluator_node.py

image

**Review by rej55 — APPROVED** [review]
LGTM

**Comment by xmfcx** [comment]
Could you rebase to main because this PR is merged:
- https://github.com/autowarefoundation/autoware.universe/pull/6495

This halves the time required for build-and-test-differential checks.

**Comment by kosuke55** [comment]
@xmfcx wow! Thank you very much!! I've been waiting for ci to finish for a long time...!

**Comment by kosuke55** [comment]
@takayuki5168 @TakaHoribe @tkimura4 
@miursh @YoshiRi @yukkysaito 
could you please approve this PR as launch code owner?

**Comment by xmfcx** [comment]
@kosuke55 I'm sorry that this job is taking forever for so many people. I am trying to speed it up as much as I can.

**Review by miursh — APPROVED** [review]

**Review by tkimura4 — APPROVED** [review]

**Pull request merged**
